HDSAviz/data_processing.py

```diff
@@ -0,0 +1,218 @@
+"""
+Tools for reading and processing the sensitivity analysis data files.
+Some of the files are specific to our project (`input_parameters.csv' and
+`results.csv`), but the results of sensitivity analyses are formatted
+as any SALib analysis results will be.
+
+Our data files are stored outside this repository because they are too large,
+so it may be necessary for future users to update the relative paths below
+(our data directory is located at ../../HDSAviz_data/
+"""
+import os
+
+import pandas as pd
+
+
+def read_file(filename, numrows=None, drop=False, sep=','):
+    """
+    Function reads a file of input parameters or model results
+    and returns a pandas dataframe with its contents.
+    The first line of the input should contain headers
+    corresponding to the column names.
+
+    Parameters:
+    ----------
+    filename  : String with the complete filename, including
+                absolute or relative path.
+    numrows   : Integer number of rows of the file to read.
+                If you don't specify this parameter all rows
+                will be read.
+    drop      : A list of strings indicating which (if any)
+                of the named columns you do not want to include
+                in the resulting dataframe. (ex. ['cats', 'dogs'],
+                default is not to drop any rows)
+    sep       : String indicating the column separator in the
+                file (optional, default = ',').
+
+    Returns:
+    --------
+    df : A pandas dataframe with the contents of the file,
+         limited to the number of rows specified and without the
+         columns named in "drop".
+    """
+    df = pd.read_csv(filename, sep=sep, nrows=numrows)
+    if not drop:
+        df.drop(drop, axis=1, inplace=True)
+
+    return df
+
+
+def get_params(numrows=None, drop=['End_time', 'Oxygen']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with all the parameters analyzed in
+    the sensitivity analysis, but not additional parameters like
+    end time and oxygen content.  If you would like all of the
+    parameters (even those not analyzed for sensitivity) then pass
+    drop=False.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which parameters you do not want to
+              include in the returned dataframe.  If you want all params
+              then pass drop=False.
+
+    Returns:
+    -------
+    A pandas dataframe
+
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
```

Comment:
# General Comment: 
We Need to Update the Directory Structure in the actual readme.md file and we need to put in some sample data files (at least a few rows).

# How about allowing user to set path by creating a set_path function which takes input from user and sets path for data folders?? 
I recommend that these functions like get_all_params(), get_params() etc. which are reading from specific path, should also have path as input, instead of only reading from ('../../HDSAviz_data/input_parameters.csv') ..... we can probably set default path as '../../HDSAviz_data/input_parameters.csv' for our case like this.
get_all_params(path='../../HDSAviz_data/input_parameters.csv', numrows=None, drop=['End_time', 'Oxygen'])